File: backend/app/Http/Resources/Escolaridad.php
<?php

namespace App\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

/*
class Escolaridades extends Model {
    protected $fillable = [
        'id',
        'nombre',
    ];
}
*/


class Escolaridad extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'id' => $this->idEscolaridad,
            'nombre' => $this->nombreEscolaridad,
        ];
    }
}

File: movil/lib/classes/Escolaridad.dart
class Escolaridades
{
  List<Escolaridad> escolaridades = new List();

  Escolaridades.fromJsonList(List<dynamic> json)
  {
    if(json == null) return;

    for(var item in json)
    {
      Escolaridad esco = new Escolaridad.fromJsonMap(item);
      escolaridades.add(esco);
      print(esco);
    }
  }
}



class Escolaridad 
{
  int idEscolaridad;
  String nombre;

  Escolaridad(this.idEscolaridad, this.nombre);
  
  Escolaridad.fromJsonMap(Map<String, dynamic> json)
  {
    this.idEscolaridad = json['id'];
    this.nombre = json['nombre'];
  }
}